osu.Game/Screens/Edit/Timing/TapTimingControl.cs

```diff
@@ -29,12 +30,15 @@ public partial class TapTimingControl : CompositeDrawable
         [Resolved]
         private Bindable<ControlPointGroup> selectedGroup { get; set; } = null!;
 
+        private readonly IBindable<Track> track = new Bindable<Track>();
+
         private readonly BindableBool isHandlingTapping = new BindableBool();
 
         private MetronomeDisplay metronome = null!;
+        private Container<WaveformComparisonDisplay> waveformContainer = null!;
 
         [BackgroundDependencyLoader]
-        private void load(OverlayColourProvider colourProvider, OsuColour colours)
+        private void load(OverlayColourProvider colourProvider, OsuColour colours, EditorClock clock)
```

Comment:
Looking back at this, it's pretty weird that we're binding through `EditorClock`.... but I guess that's for another discussion another day.

Reply:
What would the alternative be? Via `WorkingBeatmap` directly? Make the waveform a bindable or something?